File: Programmers/Lv3/network.cpp
#include <string>
#include <vector>

using namespace std;

void dfs(vector<vector<int>> computers,vector<bool> &visited,int computer,int n){
    visited[computer] = true;
    for(int i=0; i < n; i++){
        if(!visited[i] && computers[computer][i]){
            dfs(computers,visited,i,n);
        }
    }
}

int solution(int n, vector<vector<int>> computers) {
    int answer = 0;
    vector<bool> visited(n,false);
    
    for(int i=0; i < n; i++){
        if(!visited[i]){
            answer++;
            dfs(computers,visited,i,n);
        }
    }
    return answer;
}

File: java/src/programmers/lv3/Network.java
package programmers.lv3;

public class Network {
    boolean[] visited;
    public int solution(int n, int[][] computers) {
        int answer = 0;
        visited = new boolean[n];

        for(int i=0; i < n; i++){
            if(!visited[i]){
                answer++;
                visited[i] = true;
                dfs(i, computers);
            }
        }
        return answer;
    }
    private void dfs(int start,int[][] computers){
        visited[start] = true;
        for(int i = 0; i < computers.length; i++){
            if(!visited[i] && computers[start][i] == 1){
                dfs(i, computers);
            }
        }
    }
}
